astropy/table/operations.py

```diff
@@ -1224,6 +1253,58 @@ def _join(left, right, keys=None, join_type='inner',
     return out
 
 
+def _join_keys_left_right(left, right, keys, keys_left, keys_right, join_funcs):
+    """Do processing to handle keys_left / keys_right args for join.
+
+    This takes the keys_left/right inputs and turns them into a list of left/right
+    columns corresponding to those inputs (which can be column names or column
+    data values). It also generates the list of fake key column names (strings
+    of "1", "2", etc.) that correspond to the input keys.
+    """
+    def _keys_to_cols(keys, table, label):
```

Comment:
Any reason not to put it outside of the function? It doesn't seem to use any variables from the function scope.

Reply:
I thought about this. I sometimes like to localize functions this way to make them easier to find and emphasize that they are really just applicable to this one function. I think in this case there is no noticeable performance penalty.

Reply:
Sure, that makes sense too.